proposals/0095-AOA-multiplexing.md

```diff
@@ -14,161 +14,154 @@ This proposal aims to add AOA multiplexing capability to SdlRouterService.
-		 * @param messenger
-		 * @param theType
-		 */
-		public RegisteredApp(String appId, Messenger messenger, TransportType theType){
-		    ...
-			transportTypes = new HashMap<>();
-			transportTypes.put(0L, theType); // default session maps to the given (primary) TransportType
-		}
-	}
-	
-	@Override
-	public void handleMessage(Message msg) {
-	    ....    
-        switch (msg.what) {
-	        case TransportConstants.ROUTER_REGISTER_CLIENT:
-	            RegisteredApp app = service.new RegisteredApp(appId,msg.replyTo, TransportType.MULTIPLEX); // This is the case for Bluetooth. AOA's message handler specifies TransportType.MULTIPLEX_AOA
-	    }
-	}
+public void onEnableSecondaryTransport(byte sessionID, ArrayList<String> secondaryTransport,
+	        ArrayList<Integer> audio, ArrayList<Integer> video, TransportType transportType) {
+                // check to see if secondaryTransport contains "AOA_USB", and then handle it as AOA multiplexing
+	        }
+
 ```
 
-### SdlRouterService uses either MultiplexBluetoothTransport or MultiplexAoaTransport depending on the TransportType
+### SdlRouterService uses either MultiplexBluetoothTransport or MultiplexAoaTransport depending on binding action
+MultiplexAoaTransport encapsulates the actual read/write operation of AOA transport.
 ```java
 public class SdlRouterService extends Service {
     ...
-    // underlying transport depending on the TransportType.
+    // underlying transport depending on the type of Multiplexing.
 	private MultiplexBluetoothTransport bluetoothTransport;
 	private MultiplexAoaTransport aoaTransport;
     ...
+
+	@Override
+	public IBinder onBind(Intent intent) {
+        String requestType = intent.getAction();
+        if(TransportConstants.BIND_REQUEST_TYPE_CLIENT.equals(requestType)) {
+            // corresponding Binder will use bluetoothTransport
+        } else if (TransportConstants.BIND_REQUEST_TYPE_AOA_CLIENT.equals(requestType)) {
+            // corrsponding Binder will use aoaTransport
+        }
+    }
 }
 ```
 
 ### The app that has actual UsbAccessory's permission sends ParcelFileDescriptor to active SdlRouterService
-We discussed about a potential issue where multiple SDL applications have the same accessory_filter settings.
+When a user plugs in their device to an AOA connection, they are prompted with selection dialog. The dialog contains all apps that support the currently connected accessory. In this dialog there is also an option to always open a specific app for that accessory.
+We basically assume the app that receives AOA intent gets the accessory permission, and then the app works as the router for AOA transport. if the user picks an app to always receive the AOA intent and it propagates the router service, all multiplexed connections will go through that apps router service regardless if there is a newer router service present. If by chance that is not a trusted router service, no other apps will ever connect until the user clears the flag.
```

Comment:
Please update the second sentence to "If the user... will go through that app's router service regardless of if there..."